src/main/java/org/cactoos/text/JoinedText.java

```diff
@@ -85,17 +77,14 @@ public JoinedText(final Text delimit, final Text... txts) {
      * @param txts Texts to be joined
      */
     public JoinedText(final Text delimit, final Iterable<Text> txts) {
-        this.delimiter = delimit;
-        this.texts = txts;
-    }
-
-    @Override
-    public String asString() throws Exception {
-        final StringJoiner joint = new StringJoiner(this.delimiter.asString());
-        for (final Text text : this.texts) {
-            joint.add(text.asString());
-        }
-        return joint.toString();
+        super((Scalar<String>) () -> {
+            final StringJoiner joint =
+                new StringJoiner(delimit.asString());
+            for (final Text text : txts) {
+                joint.add(text.asString());
+            }
+            return joint.toString();
+        });
```

Comment:
@krzyk We should not have this join happening here in the constructor, see https://www.yegor256.com/2015/05/07/ctors-must-be-code-free.html.

Reply:
@paulodamaso this follows how the code behaves in other cases in cactoos, I'm not sure if calling a lambda that will be evaulated later (or any other class creation) is a "code". 
See this snippet from that article:

>  This is what imperative programming is about. In imperative programming, we do all calculations right now and return fully ready results. In declarative programming, we are instead trying to delay calculations for as long as possible.

Lambdas and anonymous classes are indead delaying the calculations as long as possible.

Reply:
@krzyk You are right, again. Lambdas are evaluated later than object creation, so your code is right.